modules/apps/foundation/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/service/impl/FriendlyURLEntryLocalServiceImpl.java

```diff
@@ -71,11 +73,28 @@ public FriendlyURLEntry addFriendlyURLEntry(
 			friendlyURLEntryPersistence.update(mainFriendlyURLEntry);
 		}
 
-		FriendlyURLEntry oldFriendlyURLEntry =
-			friendlyURLEntryPersistence.fetchByG_C_C_C_U(
-				groupId, companyId, classNameId, classPK, normalizedUrlTitle);
+		for (String urlTitle : urlTitleMap.values()) {
+			String normalizedUrlTitle = FriendlyURLNormalizerUtil.normalize(
+				urlTitle);
+
+			FriendlyURLEntryLocalization oldFriendlyURLEntryLocalization =
+				friendlyURLEntryLocalizationPersistence.fetchByG_C_C_C_U(
+					groupId, companyId, classNameId, classPK,
+					normalizedUrlTitle);
+
+			FriendlyURLEntry oldFriendlyURLEntry = null;
+
+			if (oldFriendlyURLEntryLocalization != null) {
+				oldFriendlyURLEntry =
+					friendlyURLEntryPersistence.fetchByPrimaryKey(
+						oldFriendlyURLEntryLocalization.
+							getFriendlyURLEntryId());
+			}
+
+			if (oldFriendlyURLEntry == null) {
+				continue;
+			}
 
-		if (oldFriendlyURLEntry != null) {
 			oldFriendlyURLEntry.setMain(true);
 
 			return friendlyURLEntryPersistence.update(oldFriendlyURLEntry);
```

Comment:
This return can not be right. I guess we are not clear about the api contract for https://github.com/shuyangzhou/liferay-portal/pull/4464/commits/7aa4805858cec34a8d608c70026c1c7ddd8595d7#diff-e54f782412436922875fd170c64ef728R59

In case there are multiple translation in the urlTitleMap, and the first map entry matches an exist FriendlyURLEntry, then it will trigger this return, which will basically ignore every other entry in the map. That does not make sense.

Either the "add" should be rejected in case a previous matching FriendlyURLEntry exist, or the "add" should behave as "update" on seeing a previsously exist FriendlyURLEntry.
Right now, the "add" will mark the previous matching FriendlyURLEntry as "active" one, but throw away everything in that urlTitleMap. This can not be a right behavior.